stripped/colabs/GrupowanieEM_stripped.py

```diff
@@ -0,0 +1,396 @@
+     labels1[j] = labels1[j] + labels2[j]
+
+for j in range(len(labels1)):
+  if labels1[j]>=4:
+     labels1[j] = 1
+  else:
+     labels1[j] = 0
+
+accuracy_score(labels1, labels_)
+
+precision_score(labels1, labels_)
+
+jaccard_score(labels1, labels_)
+
+NASA2020_full = nasa2020.toPandas()
+
+#Srednie dla wybranych cech:
+NASA2020_means = (NASA2020_full[['lon', 'lat', 'Evap','PotEvap','RootMoist','Rainf','SoilM_100_200cm', 'GVEG']].groupby(
+    by=['lon', 'lat']).mean()).reset_index()
+
+#Roczna suma opadów:
+NASA2020_sum = (NASA2020_full[['lon', 'lat', 'Rainf']].groupby(by=['lon', 'lat']).sum()).rename(columns={"Rainf": "Annual Rainfall"}).reset_index()
+
+NASA2020 = NASA2020_means.merge(NASA2020_sum, how='inner', on=['lon', 'lat'])
+
+#Mediany dla wybranych cech:
+NASA2020_median = (NASA2020_full[['lon', 'lat', 'PotEvap', 'Evap', 'SoilM_100_200cm', 'AvgSurfT', 'SoilT_40_100cm' ]]).rename(
+    columns={'PotEvap': "PotEvap_Median", 'Evap': 'Evap_Median', 'AvgSurfT': 'AvgSurfTmedian', 'SoilT_40_100cm': "SoilT40_100_Median",
+            'SoilM_100_200cm': 'SoilM_100_200cm_Median'}).groupby(by=['lon', 'lat']).median().reset_index()
+
+NASA2020 = NASA2020.merge(NASA2020_median, how='inner', on=['lon', 'lat'])
+
+
+for i in range(4, 10):
+    NASA2020 = add_column('RootMoist', i , NASA2020_full, NASA2020)
+
+for i in range(4, 11):
+    NASA2020 = add_column('GVEG', i , NASA2020_full, NASA2020)
+
+for feat in ['Rainf', 'PotEvap', 'Evap', 'Albedo']:
+   for i in range(5, 10):
+
+     NASA2020 = add_column(feat, i, NASA2020_full, NASA2020)
+
+
+NASA2020.head()
+
+NASA2020feat = NASA2020.drop(columns=['lon', 'lat'])
+
+#standaryzujemy dane:
+scaler = preprocessing.StandardScaler()
+scaler.fit(NASA2020feat)
+standarized = scaler.transform(NASA2020feat)
+
+st_NASA2020 = pd.DataFrame(standarized, columns=NASA2020feat.columns)
+st_NASA2020.head()
+
+gm = GaussianMixture(n_components = 2, n_init = 200, max_iter=200, init_params= 'random_from_data', covariance_type='spherical', random_state=22)
+gm_result = gm.fit_predict(st_NASA2020)
+
+gm_list = gm_result.tolist()
+number_of_label1 = sum(gm_list)
+print(number_of_label1)
+
+percentage_deserts_predicted = sum(gm_list)/len(NASA2020)
+print(percentage_deserts_predicted)
+
+NASA2020['label'] = gm_list
+anotated_unique = NASA_2020_set[NASA_2020_set['Month']==1]
+anotated_subset = NASA2020[['lon', 'lat', 'label']].merge(anotated_unique[['lon', 'lat', 'pustynia']], on=['lon', 'lat'], how='inner')
+
+accuracy_score(list(anotated_subset['label']), list(anotated_subset['pustynia']))
+
+precision_score(list(anotated_subset['label']), list(anotated_subset['pustynia']))
+
+recall_score(list(anotated_subset['label']), list(anotated_subset['pustynia']))
+
+def GM_monthly_data(month: int, df: pd.DataFrame) -> Tuple[List[int], float, float, float]:
+
+  NASA_M_2020 = df[df['Month']==month]
+  NASA_month = NASA_M_2020.drop(columns=['lon', 'lat', 'Month', 'SoilT_40_100cm', 'SoilM_100_200cm'])
```

Comment:
czemu soil wyleciały?

Reply:
Dwa zmienne w naszym datasecie związane są z temperaturą (jeszcze AvgSurfT), pomyślałam więc, żeby jedną z nich na próbę wyrzucić - i rzeczywiście troszkę to poprawiło wyniki (nie pamiętam już o ile)